File: DisappearedNumbersInArray.java
// Time Complexity : O(n)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : None

public class DisappearedNumbersInArray {
    class Solution {
        public List<Integer> findDisappearedNumbers(int[] nums) {
            int n = nums.length;
            List<Integer> list = new ArrayList<>();
            for(int i = 0; i < n; i++){
                int val = Math.abs(nums[i]);

                if(nums[val - 1] > 0){
                    nums[val - 1] *= -1;
                }
            }

            for(int i = 0; i < n; i++){
                if(nums[i] < 0){
                    nums[i] *= -1;
                }
                else{
                    list.add(i + 1);
                }
            }

            return list;
        }
    }
}

File: disappearedNumbersInArray.py
# Space Complexity:  O(1)
# TIme Complaxity : O(N)
from typing import List

class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        res = []
        for idx, num in enumerate(nums):
            temp = abs(num)
            if nums[temp-1] > 0:
                nums[temp-1] = -1*nums[temp-1]
        
        for idx, num in enumerate(nums):
            if num < 0:
                nums[idx] = -1*nums[idx]
            else:
                res.append(idx+1)
        
        return res
